skiko/src/jvmMain/kotlin/org/jetbrains/skiko/SkiaLayerProperties.kt

```diff
@@ -1,6 +1,22 @@
 package org.jetbrains.skiko
 
-internal data class SkiaLayerProperties(
+
+/**
+ * SkiaLayerProperties is a data class that represents the rendering configuration for a SkiaLayer.
+ *
+ * @property isVsyncEnabled Specifies whether vertical synchronization (VSync) is enabled.
+ * Default value is [SkikoProperties.vsyncEnabled]. Setting this to true is a hint toward underlying implementation
+ * to synchronize the rendering with the display presentation. It guarantees that the frame is presented without
+ * visual artifacts like tearing in exchange for a possible latency increase.
+ * @property isVsyncFramelimitFallbackEnabled Specifies whether framelimit fallback is enabled (software renderer).
+ * Default value is [SkikoProperties.vsyncFramelimitFallbackEnabled].
+ * @property renderApi Specifies the graphics API used for rendering.
+ * Default value is [SkikoProperties.renderApi].
+ * @property adapterPriority Specifies the GPU that will be selected for rendering.
+ * Default value is [SkikoProperties.gpuPriority].
+ */
+@ExperimentalSkikoApi
+data class SkiaLayerProperties(
```

Comment:
Data classes should not be a part of public API. See [Avoid using data classes in your API﻿](https://kotlinlang.org/docs/api-guidelines-backward-compatibility.html#avoid-using-data-classes-in-your-api)

Reply:
Sorry, it was considered a draft PR for internal discussion.
Not relevant anymore, the class stays internal.

Reply:
Updated